WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -516,13 +516,25 @@ final class ProductFormViewController<ViewModel: ProductFormViewModelProtocol>:
                 guard isEditable else {
                     return
                 }
-                // TODO: 11090 - Analytics
+
+                if product is EditableProductVariationModel {
+                    ServiceLocator.analytics.track(.productVariationViewSubscriptionFreeTrialTapped)
+                } else if product is EditableProductModel {
+                    ServiceLocator.analytics.track(.productDetailsViewSubscriptionFreeTrialTapped)
+                }
```

Comment:
Instead of checking the product model's type, we should be using `eventLogger` property from this view controller to log these events. [Example code](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/ViewRelated/Products/Edit%20Product/ProductFormViewController.swift#L397)